### ISSUE TITLE
Server IP is blank on every reboot

### ISSUE THREAD

**Issue by geraldvillorente** [debauchee/barrier#593]
### Operating Systems ###

Server: MacOS Catalina

Client: Ubuntu 18.04 LTS

This is related to this closed issue https://github.com/debauchee/barrier/issues/44

Every time I reboot the client the Server IP is blank.

### Barrier Version ###

2.3.2

### Steps to reproduce bug ###

1. Setup the server on macOS.
2. Setup the client on Ubuntu. 
3. At this point, the Barrier should be working fine.
4. Reboot the client.
5. Confirm that the Server IP field is empty.

**Comment by maluramichael** [comment]
Try the following: Open the client on the guest. Put in the ip, in the menu press `barrier -> save` configuration, choose a save location and name, close the client without pressing `start`.

Eventually i got it to work this way.

**Comment by geraldvillorente** [comment]
Thank you @maluramichael. I'll try this.

**Comment by AlexBSM** [comment]
This seemed to work for me: 
1) open a new Konsole in ~/usr/local/bin
2) run sudo ./barrier in the above mentioned folder (important as you will need sudo privileges to create a file in this folder)
3) add your server ip and any other setting on the barrier gui
4) use the barrier tab on the gui to "save configuration" now save it to ~/usr/local/bin and call it ".barrier.sgc" (the sgc part is super important)
5) Profit

**Comment by geraldvillorente** [comment]
@AlexBSM awesome! This fixed the issue I am having.

**Comment by martinpescador** [comment]
This workaround works on one device for me, but not on another?!?

**Comment by crazydef** [comment]
I copied the `.barrier.sgc` file to `~/.local/bin` on Linux Mint and that worked for me.

**Comment by sekmenhuseyin** [comment]
Hi
I have the same problem, server IP is not saved. I've tried a dozen times with different settings
![image](https://github.com/debauchee/barrier/assets/1363876/af11b768-9f6f-4592-b754-e36c448c304f)

**Comment by myviolinsings** [comment]
Am also having this issue after fresh install of Linux Mint using Gnome on Surface Pro 7. Everything else works except barrier doesn't start on login and it never loads saved conf file. When I tried the above in terminal at /usr/local/bin$ sudo ./barrier, I get "command not found." And saving config does not appear to save file as .sgc, but rather just a text file. A search where I saved it in home for *.sgc, including typing the name of the file I can see there, does not find any .sgc files. If I type the IP address and hit start, it continues to say, "Barrier is starting" yet the mouse & keyboard start working.
